File: mobile-engage/src/androidTest/java/com/emarsys/mobileengage/testUtil/CollectionTestUtils.java
package com.emarsys.mobileengage.testUtil;

import java.util.List;
import java.util.NoSuchElementException;

public class CollectionTestUtils {
    public static int numberOfElementsIn(List list, Class type) {
        int count = 0;

        for (Object object : list) {
            if (object.getClass().equals(type)) {
                count++;
            }
        }

        return count;
    }

    public static int numberOfElementsIn(Object[] array, Class type) {
        int count = 0;

        for (Object object : array) {
            if (object.getClass().equals(type)) {
                count++;
            }
        }

        return count;
    }

    public static <T> T getElementByType(List<?> list, Class<T> type) {
        for (Object o : list) {
            if (type.isInstance(o)) {
                return type.cast(o);
            }
        }
        throw new NoSuchElementException("Cannot find element of class " + type + " in " + list);
    }
}

File: testUtils/src/main/java/com/emarsys/testUtil/CollectionTestUtils.kt
package com.emarsys.testUtil

import java.util.*

object CollectionTestUtils {

    @JvmStatic
    fun numberOfElementsIn(list: List<Any?>, type: Class<*>): Int {
        return list
                .filterNotNull()
                .filter { it::class.java == type }
                .count()
    }

    @JvmStatic
    fun numberOfElementsIn(array: Array<Any>, type: Class<*>): Int {
        var count = 0

        for (o in array) {
            if (o.javaClass == type) {
                count++
            }
        }

        return count
    }

    @JvmStatic
    fun <T> getElementByType(list: List<*>, type: Class<T>): T? {
        for (o in list) {
            if (type.isInstance(o)) {
                return type.cast(o)
            }
        }
        throw NoSuchElementException("Cannot find element of class $type in $list")
    }
}
